Sources/Transition/HeroTransition+Start.swift

```diff
@@ -46,9 +46,18 @@ extension HeroTransition {
     }
 
     // take a snapshot to hide all the flashing that might happen
-    fullScreenSnapshot = transitionContainer?.window?.snapshotView(afterScreenUpdates: true) ?? fromView?.snapshotView(afterScreenUpdates: true)
-    if let fullScreenSnapshot = fullScreenSnapshot {
+    if #available(iOS 11.0, *) {
+     DispatchQueue.main.async {
```

Comment:
Why do we need `DispatchQueue.main.async`?

Reply:
I'm trying to find out a good explaination... :)
Normally it takes a snapshot with dark alpha on it. With main.async or just changing afterScreenUpdates to false it works. `fullScreenSnapshot = transitionContainer?.window?.snapshotView(afterScreenUpdates: false) ?? fromView?.snapshotView(afterScreenUpdates: false)`

Reply:
One problem is that the main thread could potentially block and delay your snapshot capture time. If that happens, the snapshot won't be able to hide the flash anymore.

Before your patch, does `window?.snapshotView(afterScreenUpdates: )` capture a nil view? I haven't look into this but it that the reason why the flash happens in the first place?

Reply:
Nope, there is a view

Reply:
Yeah, you're right. It takes out some performance. But taking a snapshot before screen updates improves it. I'll test it